include/deal.II/base/quadrature.h

```diff
@@ -221,6 +221,19 @@ class Quadrature : public Subscriptor
   template <class Archive>
   void serialize (Archive &ar, const unsigned int version);
 
+  /**
+   * This function returns true if the quadrature object is a tensor product
+   * of one-dimensional formulas.
+   */
+  bool is_tensor_product() const;
+
+  /**
+   * In case the quadrature formula is a tensor product, this function
+   * returns the one-dimensional basis objects.
+   * Otherwise, calling this function is not allowed.
+   */
+  const std::vector<Quadrature<1>> &get_tensor_basis() const;
```

Comment:
This is difficult. `Quadrature<1>` can't have a member variable of type `std::array<Quadrature<1> >`. Since `std::array` needs to know the size of this object which is not known. This would probably also end in an infinite loop. The reason why `std::vector<Quadrature<1> > ` can be used here is the fact that it rather uses dynamic memory allocation.

Consequently, I can't return a const reference to such a `std::array` in the one-dimensional case. If we return a copy in `get_tensor_basis` or don't implement this function for the one-dimensional case, we can get around this. would you prefer this?